pymeasure/instruments/common_base.py

```diff
@@ -0,0 +1,569 @@
+                 fset_params_list=None, prefix=""):
+        super().__init__(fget, fset, fdel, doc)
+        self.fget_params_list = () if fget_params_list is None else fget_params_list
+        self.fset_params_list = () if fset_params_list is None else fset_params_list
+        self.name = ""
+        self.prefix = prefix
+
+    def __get__(self, obj, objtype=None):
+        if obj is None:
+            # Property return itself when invoked from a class
+            return self
+        if self.fget is None:
+            raise AttributeError(f"Unreadable attribute {self.name}")
+
+        kwargs = {}
+        for attr in self.fget_params_list:
+            attr_instance_name = self.prefix + "_".join([self.name, attr])
+            if hasattr(obj, attr_instance_name):
+                kwargs[attr] = getattr(obj, attr_instance_name)
+        return self.fget(obj, **kwargs)
+
+    def __set__(self, obj, value):
+        if self.fset is None:
+            raise AttributeError(f"Can't set attribute {self.name}")
+        kwargs = {}
+        for attr in self.fset_params_list:
+            attr_instance_name = self.prefix + "_".join([self.name, attr])
+            if hasattr(obj, attr_instance_name):
+                kwargs[attr] = getattr(obj, attr_instance_name)
+        self.fset(obj, value, **kwargs)
+
+    def __set_name__(self, owner, name):
+        self.name = name
+
+
+class CommonBase:
+    """Base class for instruments and channels.
+
+    This class contains everything needed for pymeasure's property creator
+    :meth:`control` and its derivatives :meth:`measurement` and :meth:`setting`.
+    """
+
+    # Variable holding the list of DynamicProperty parameters that are configurable
+    # by users
+    _fget_params_list = ('get_command',
+                         'values',
+                         'map_values',
+                         'get_process',
+                         'command_process',
+                         'check_get_errors')
+
+    _fset_params_list = ('set_command',
+                         'validator',
+                         'values',
+                         'map_values',
+                         'set_process',
+                         'command_process',
+                         'check_set_errors')
+
+    # Prefix used to store reserved variables
+    __reserved_prefix = "___"
+
+    def __init__(self):
+        self._special_names = self._setup_special_names()
+        # Add children from ChildDescriptors.
+        for item, value in inspect.getmembers(self.__class__):
+            if isinstance(value, ChildDescriptor):
+                if isinstance(value.id, (list, tuple)):
+                    for id, cls in zip(value.id, value.cls):
+                        self.add_child(cls, id, collection=item, **value.kwargs)
+                else:
+                    self.add_child(value.cls, value.id, collection=item, **value.kwargs)
+
+    @staticmethod
+    def children(id, cls, **kwargs):
```

Comment:
```suggestion
    def children(cls, id=None, **kwargs):
```
If you reverse those (also where used, of course), you can deal with the case of `id=None` naturally.

Reply:
Done.